fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysis.kt

```diff
@@ -347,55 +373,93 @@ private enum class Highlight(
     ),
 }
 
+private class ProductAnalysisPreviewModel(
```

Comment:
I see the need for this, but throwing an idea out there, would be really nice and convenient to use the same state model with fake data for tests and previews.

Reply:
I think that makes sense. When I make the above changes, I'll also look into refactoring this to see how/if we can parallelize the previews and tests.